app/utils/AccountUtils.scala

```diff
@@ -17,18 +17,26 @@
 package utils
 
 import controllers.auth.AuthenticatedRequest
-import models.ActingAsAttorneyFor
+import models.{ActingAsAttorneyFor, AgentToken}
 import play.api.i18n.Messages
 import play.api.mvc.Request
 import uk.gov.hmrc.domain.{SaUtr, TaxIdentifier, Uar}
 
 trait AccountUtils {
-  def getAccount(request: AuthenticatedRequest[_]): TaxIdentifier =
-    request.agentRef.getOrElse(request.saUtr.getOrElse(SaUtr("")))
+  def getAccount(request: AuthenticatedRequest[_]): Option[TaxIdentifier] =
+    if (request.agentRef.isDefined) {
+      Some(request.agentRef.get)
+    } else if (request.saUtr.isDefined) {
+      Some(request.saUtr.get)
+    } else {
+      None
+    }
+
   //This warning is unchecked because we know that AuthorisedFor will only give us those accounts
   def getAccountId(request: AuthenticatedRequest[_]): String = (getAccount(request): @unchecked) match {
-    case sa: SaUtr => sa.utr
-    case ta: Uar   => ta.uar
+    case Some(sa: SaUtr) => sa.utr
+    case Some(ta: Uar)   => ta.uar
+    case _ => ""
```

Comment:
This is confusing, you had this as an Option before and now it's an empty string? Keep it as an Option rather than pass a String type around. It will simplify your code